### ISSUE TITLE
Rule 'object-shorthand' with es7.objectRestSpread

### ISSUE THREAD

**Issue by AlexKVal** [eslint/eslint#2486]
The project [react-bootstrap](https://github.com/react-bootstrap/react-bootstrap) uses [es7.objectRestSpread](https://github.com/sebmarkbage/ecmascript-rest-spread)

I've got three warnings
```sh
src/Button.js
  34:6  warning  Expected property shorthand  object-shorthand

src/Input.js
  181:20  warning  Expected property shorthand  object-shorthand

src/NavItem.js
  32:8  warning  Expected property shorthand  object-shorthand
```
For [src/Button.js#L34](https://github.com/react-bootstrap/react-bootstrap/blob/6dfcf3688eea38/src/Button.js#L34)
```js
classes = {
  active: this.props.active,
  'btn-block': this.props.block,
  ...classes  // this line
};
```

For [src/Input.js#L181](https://github.com/react-bootstrap/react-bootstrap/blob/e694e00ccec612/src/Input.js#L181)
```js
let {bsStyle, ...other} = this.props; /* eslint no-unused-vars: 0 */
```

For [src/NavItem.js#L32](https://github.com/react-bootstrap/react-bootstrap/blob/6dfcf3688eea38/src/NavItem.js#L32)
```js
let {
    disabled,
    active,
    href,
    title,
    target,
    children,
    ...props } = this.props; // this line
```

**Comment by ilyavolodin** [comment]
Duplicate of #2352. We do not support experimental features at this point. If you would like to use eslint with the code that uses es7 proposed features, you need to switch parser to [`babel-eslint`](https://github.com/babel/babel-eslint).

**Comment by ilyavolodin** [comment]
Actually, you are most likely already using `babel-eslint`, otherwise you would get parsing errors, instead of breaking rules. But we still don't support es7 features, so I think your best bet would be to copy `object-shorthand` rule, modify it, and package it as a plugin.

**Comment by AlexKVal** [comment]
> we don't support es7 features

Yeah. I have assumed the same in our issue.

Thank you for the answer. :+1: 
I close this issue.
(unless you have somewhere ES7 features hidden Milestone :smile: )

> .. copy object-shorthand rule, modify it, and package it as a plugin.

Maybe `babel-eslint` should monkeypatch this rule ?
I'm going to look through the `babel-eslint` code-base to find out answer for this.

I have missed `babel-eslint` from my view. Thank you for pointing out to it :sparkles:
